### PR TITLE
Added support for update validators

### LINKED ISSUE #197: Support update validators
Add `validate_doc_update`  property accessor in design document module for update validators.
Subitem part of #137.

### PR DESCRIPTION
**Pull request by emlaver**
## What

Support update validators with ``validate_doc_update`` property in design document module.
Documentation and an example are included with the property accessor.

## How

- Added ``validate_doc_update`` property with documentation

## Testing

Added unit test in DesignDocumentTests.

## Issues
fixes #197

### CODE REVIEW

File: src/cloudant/design_document.py [hunk 1]
```diff
@@ -49,6 +49,36 @@ def __init__(self, database, document_id=None):
             self.setdefault(prop, dict())
 
     @property
+    def validate_doc_update(self):
+        """
+        Provides an accessor property to the update validators dictionary in
+        the locally cached DesignDocument.  Update validators evaluate whether a
+        document should be written to disk when insertions and updates are attempted.
+
+        Note: Update validators do not require a query because they implicitly
```
Comment (ricellis) [inline src/cloudant/design_document.py]: I don't think we should say this. It makes it sound like accessing the property runs the validator.

Comment (emlaver) [inline src/cloudant/design_document.py]: Removed this note in c7122d7.

**Comment by rhyshort** [comment]
+1

**Comment by ricellis** [comment]
+1

**Pull request merged**